src/Components/Web/src/Forms/EditForm.cs

```diff
@@ -73,11 +83,16 @@ public EditForm()
         /// <inheritdoc />
         protected override void OnParametersSet()
         {
-            if ((EditContext == null) == (Model == null))
+            if (_hasSetEditContextExplicitly && Model != null)
             {
                 throw new InvalidOperationException($"{nameof(EditForm)} requires a {nameof(Model)} " +
                     $"parameter, or an {nameof(EditContext)} parameter, but not both.");
             }
+            else if (EditContext == null && Model == null)
```

Comment:
Should this now be `else if (!_hasSetEditContextExplicitly && Model == null)` ?

Because the rule is: you must supply a model if and only if you didn't supply an edit context.

In the event where you:

 * First supplied a non-null model
 * Then on the next render supplied a null model

Then we'd want to throw I think.

Reply:
Yes, agreed. I'll update it.